std/polynomial/polynomial.go

```diff
@@ -4,13 +4,78 @@ import (
 	"math/bits"
 
 	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark-crypto/utils"
 )
 
 type Polynomial []frontend.Variable
 type MultiLin []frontend.Variable
 
 var minFoldScaledLogSize = 16
 
+func FromSlice(s []frontend.Variable) []*frontend.Variable {
+	r := make([]*frontend.Variable, len(s))
+	for i := range s {
+		r[i] = &s[i]
+	}
+	return r
+}
+
+// FromSliceReferences maps slice of emulated element references to their values.
+func FromSliceReferences(in []*frontend.Variable) []frontend.Variable {
+	r := make([]frontend.Variable, len(in))
+	for i := range in {
+		r[i] = *in[i]
+	}
+	return r
+}
+
+func _clone(m MultiLin, p *Pool) MultiLin {
```

Comment:
I don't think we need it. Imo these functions were there for the prover which works on the machine-native (not circuit-native) values.